feat(i18n): domain routing strategy (#5469)
withastro/docs: src/content/docs/en/guides/internationalization.mdx

- URLs without a locale prefix, (e.g. `example.com/blog/`) will return a 404 (not found) status code.

#### `'domain'`

Use this strategy when you need to support multiple domains (e.g. `fr.example.com/blog`, `example.es/blog`). 

This routing strategy is available only for pages that are [rendered on-demand](/en/core-concepts/rendering-modes/#on-demand-rendered).

> You must also specify the URL domain pattern to use in a `domains` configuration object. For any `locales` not configured in `domains`, the fallback URLs created will follow [`prefix-other-locales`](#prefix-other-locales), and create a `/[locale]/` path for all non-default languages.

Reviewer: ```suggestion
You must also specify the URL domain pattern to use in a `domains` configuration object. For any `locales` not configured in `domains`, a `/[locale]/` path for all non-default languages will be created.
```
